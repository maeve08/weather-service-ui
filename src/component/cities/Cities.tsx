import * as React from 'react';

import {observable} from 'mobx';
import {observer} from 'mobx-react';
import {
  default as appState,
  AppState
} from '../../state/appState';
import './cities.css';

interface CitiesPageProp {
  appState: AppState;
}

@observer
export class Cities extends React.Component<CitiesPageProp> {
  props;

  @observable
  cities;

  constructor(props: CitiesPageProp) {
    super(props);

    appState.citiesState.getCitiesResponse();

  }

  render() {
    return (

      <div id={'pageContainerId'}>

        <h1>Temperatures per City</h1>
              <div id={'tableContainerId'}>
                <table>
                  <thead>
                    <tr>
                      <td>City</td>
                      <td>Temperature</td>
                    </tr>
                  </thead>
                  <tbody>

                  { appState && appState.cities &&
                    appState.cities.map(
                      (city, index) => (
                      <tr key={index}>
                        <td>{city.name}</td>
                        <td>{city.temperature}</td>
                      </tr>
                      )
                    )
                  }

                  </tbody>
                </table>
              </div>
      </div>
    );
  }
}

export default Cities;
