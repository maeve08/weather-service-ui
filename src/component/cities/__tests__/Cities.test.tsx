import * as React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Cities
  from '../Cities';


describe('Cities Component', () => {

  beforeEach(() => {});

  test('snapshot loading pagination', () => {

    const component = shallow(
      <Cities/>
    );
    expect(toJSON(component)).toMatchSnapshot();
  });
});
