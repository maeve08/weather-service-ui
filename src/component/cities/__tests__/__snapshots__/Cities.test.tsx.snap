// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities Component snapshot loading pagination 1`] = `
<div
  id="pageContainerId"
>
  <h1>
    Temperatures per City
  </h1>
  <div
    id="tableContainerId"
  >
    <table>
      <thead>
        <tr>
          <td>
            City
          </td>
          <td>
            Temperature
          </td>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
